harvesters/src/main/java/org/fao/geonet/kernel/harvest/HarvestManagerImpl.java

```diff
@@ -124,11 +114,29 @@ public void init(ServiceContext context, boolean isReadOnly) throws Exception {
         applicationContext = context.getApplicationContext();
 
         this.readOnly = isReadOnly;
-        Log.debug(Geonet.HARVEST_MAN, "HarvesterManager initializing, READONLYMODE is " + this.readOnly);
+        Log.debug(Geonet.HARVEST_MAN,
+            String.format("HarvesterManager initializing, readonly: %s, refresh interval: %d min",
+                this.readOnly, harvesterRefreshIntervalMinutes));
         xslPath = context.getAppPath().resolve(Geonet.Path.STYLESHEETS).resolve("xml/harvesting/");
         AbstractHarvester.getScheduler().getListenerManager().addJobListener(
             HarversterJobListener.getInstance(this));
 
+
+        initialiseHarvesters(context);
+        if (harvesterRefreshIntervalMinutes > 0) {
+            // ... and schedule a periodic refresh of the configuration
+            // in case a non harvesting node modify the harvester config.
+            long now = System.currentTimeMillis();
+            long nextSecond = (now / 1000) * 1000 + 1000;
+            long toSleep = nextSecond - now + 500;
+
+            Log.debug(Geonet.HARVEST_MAN, String.format("Artificially sleeping to not refresh 'on' the minute for %s.", toSleep));
+            Thread.sleep(toSleep);
```

Comment:
Can you explain this better, I'm not really sure about the usage of `Thread.sleep`

Reply:
We were discussing (Mathieu Chaussier, can´t seem to link him here for some reason) about potential issues that could arise when removing and rescheduling the jobs. When a job would be scheduled to run at minute `1:00`, at the same time it was removed and readded, we would potentially lose that job's execution. Instead, we schedule the refreshing of the jobs 'in between' the minutes, making sure we don't interfere with running jobs. 

It's probably solving an edge case, and potentially not even applicable depending on how the scheduler functions internally, but we thought to be better safe than sorry. And for sure it's an intermediate solution, pending an implementation of e.g. messagequeues to handle the job scheduling.

Reply:
One concern raised here is that we are sleeping during the GeoNetwork startup phase but this will only happen on the node taking care of harvesting, not on others right ? (and single node GeoNetwork setup will not be affected by that)

To try to clarify, the point here is to shift the job taking care of refreshing the configuration with harvesting jobs right ? So if a harvester starts at 10:00 the config refresh will happen a bit later.

Reply:
The sleep only serves to shift the start of the 'refreshHarvesters' away from a 'clean second', and attempt to start it 'in between seconds'.

Example: 
- geonetwork starts up, and hits line 129 at millisecond `1020`
- we calculate the next second, `2000`, take the difference between them, and add `500`
- now we know how long to sleep to land 'in between' seconds, with +- `500` as the millisecond part of the timer

When scheduling the refreshHarvesters, it will take the 'millisecond' part from 'currenttimemillis', which is why we have to do this manipulation. The regular jobs that are defined will most likely always start 'on the second', so well away from the refreshHarvester modification. The sleep should always be <500ms, I guess, so negligible.

Reply:
Thanks for all the clarifications @joachimnielandt